terraref/laser3d/laser3d_stats.py

```diff
@@ -0,0 +1,63 @@
+from osgeo import gdal
+import numpy as np
+import matplotlib.pyplot as plt
+import rpy2.robjects as robjects
+import rpy2.robjects.numpy2ri
+from rpy2.robjects import pandas2ri
+
+
+class GeotiffStats(object):
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.file = gdal.Open(file_path)
+        self.vector = np.array(self.file.GetRasterBand(1).ReadAsArray())
+
+    def sample_geotiff(self, sample_num = 1000):
```

Comment:
We need to make sure that the statistics are independent of spatial autocorrelation. Perhaps we should have a quasi-random sample? I expect that the spatial autocorrelation will have a lag at the scale of leaf structure (<10cm?).

Some of the statistics can be more reliably estimated from a sample of a given size (e.g. mean can be more accurately estimated than variance or quantiles from a sample of 1000).